client/src/components/Forums/ThreadDetails/ThreadDetails.tsx

```diff
@@ -41,17 +43,30 @@ const ThreadDetail: React.FC<ThreadDetailProps> = ({ forumId, threadId }) => {
     setCreatingPost(!creatingPost);
   };
 
+  const handleDeletePost = async (postId: string) => {
```

Comment:
A larger ticket could be to write Redux thunk logic for functionalities like this. Essentially dispatch an action and reducers handle the async call.